Runn-Fast/deploy-to-heroku-action
# Always restart dynos after deploy

Closes #82

Add logic to check what dynos are currently running and decide whether to boot them (scale=1) or restart them.

This still lets production to run at scale of >=1, because it will already be running -- so it always go with the restart option.

- Also updates all dependencies to their latest versions.

## Testing

**Unit Tests**

There are unit tests for the `bootOrRestartProcess` function included in the PR.

**Manual Tests**

It is quite easy to manually test this, it just takes a while while waiting for the actions to run.

I've already manually tested this quite thoroughly on this PR: https://github.com/Runn-Fast/runn/pull/6361

The process I used is:

1. Create a branch from `runn-fast/runn#development` that uses this version of the github action (currently `1.17.0-beta.1`
2. Create a PR for that branch so that it deploys the apps to Heroku.
3. Check the action logs (click "Details" in the PR, specifically "App" job)
![image](https://user-images.githubusercontent.com/1140167/125728171-9ae8b935-7328-4146-b7aa-d143bba1c314.png)
4. Open up the "Deploy to Heroku" step and look at the last few commands.
![image](https://user-images.githubusercontent.com/1140167/125728286-382f799e-395a-4e25-8f0b-4d1eb42860d3.png)
Note the `/usr/local/bin/heroku ps:scale web=1 worker=1 --app runn-pr-6361-app` line, this means that the process was "booted" (scaled from to 0).
5. Check that the Ap and Hasura servers are running
6. Try re-running the action, this time it should restart the service (`/usr/local/bin/heroku ps:restart web --app runn-pr-6361-app`)
![image](https://user-images.githubusercontent.com/1140167/125728453-5d5422cc-0438-42a3-acb1-f9d06a387179.png)
